codaco/Network-Canvas
# Feature/remote protocol import

Replaces: https://github.com/codaco/Network-Canvas/pull/299

This feature allows us to import a zipped protocol bundle in the format:

```
myprotocol.canvas/
myprotocol.canvas/protocol.json
myprotocol.canvas/custom.js
myprotocol.canvas/assets/
myprotocol.canvas/assets/example.jpg
myprotocol.canvas/assets/...
```

There's still more work to go to support the external csv feature, but this is mainly an implementation detail now, as this change includes a HOC that can be used to load assets.

# Testing

Buttons only appear on the Electron, and Cordova Setup screens.

1. Click "Import demo protocol"

or

1. Type url of known protocol in submit box and click submit!

# Known issues

- ~It currently works for a fixed archive which exists it `public/demo.canvas`, the idea being that once we have the server distribution working it will load from some kind of temporary directory on the device. (ed: unless you load a remote protocol)~
- Very basic testing targeted at Electron only.
- In order to allow loading of remote protocols on Android, we have to allow-navigation https://*, this presents security risk in that the page might be redirected elsewhere to a spoof version of the app.


Partially resolves #238
